pkg/reconcilers/deployment_test.go

```diff
@@ -0,0 +1,89 @@
+package reconcilers_test
+
+import (
+	"github.com/kuadrant/limitador-operator/pkg/reconcilers"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Deployment", func() {
+	var desired *appsv1.Deployment
+
+	BeforeEach(func() {
+		desired = &appsv1.Deployment{
```

Comment:
nitpick: this approach is not wrong *perse*. However I find less robust than defining a function to return the object

```go
func myFancyDeployment() *appsv1.Deployment {
    return &appsv1.Deployment{ 
             ....
    }
}
```

"Robust" meaning less prone to errors like forgetting to set the new value or being `nil` for whatever reason.

Reply:
I don't see this to be less robust. The beforeEach ensures the desired deployment is set before each test context and a test context has to actively change the deployment if the test context requires some else. I find if there is a function that returns a default set up (deployment in this case) people tend to start adding logic to the function to handle different cases which kinda makes it pointless to have a default.

I am not sure what the ask is here. If you want me to rewrite the test to suit go test as in the other comment this would be gone so nothing is needed. But if the rewrite is not need I am not sure what change you are asking me to do.

Reply:
no changes requested. Just a comment to refactor to something else. Since you do not agree with me, as I said this is not wrong, thus nothing to be done here.